### PR TITLE
Update unsafe code build tool to only run when contextually relevant

### PR DESCRIPTION
**Pull request by keveleigh**
## Overview
The build configuration added in #5891 sometimes ran in contexts where it shouldn't, like IL2CPP builds where additional C# projects aren't generated.

I wrapped all calls, including making the checkbox visible, in a check for both .NET as the scripting backend and Unity C# Projects enabled.

## Changes
- Fixes: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/6308

### CODE REVIEW

File: Assets/MixedRealityToolkit/Utilities/BuildAndDeploy/UwpPlayerBuildTools.cs [hunk 1]
```diff
@@ -86,7 +86,14 @@ async void PostBuildAction(IBuildInfo innerBuildInfo, BuildReport buildReport)
                     var uwpBuildInfo = innerBuildInfo as UwpBuildInfo;
                     Debug.Assert(uwpBuildInfo != null);
                     UwpAppxBuildTools.AddCapabilities(uwpBuildInfo);
-                    UwpAppxBuildTools.UpdateAssemblyCSharpProject(uwpBuildInfo);
+
+#if !UNITY_2019_1_OR_NEWER
+                    if (EditorUserBuildSettings.wsaGenerateReferenceProjects &&
```
Comment (wiwei) [inline Assets/MixedRealityToolkit/Utilities/BuildAndDeploy/UwpPlayerBuildTools.cs]: Nit: It feels weird for us to double guard this (i.e. once prior to invocation time, and again within the function itself. It feels like the function should either handle it for all callers (i.e. it abstracts out that complication) or it should define formally that it's responsibilities don't include checking, so callers should check.

**Comment by wiwei** [comment]
@Weasy666 FYI in case you wanna try this out!

**Comment by Weasy666** [comment]
@wiwei thanks for the info.
@keveleigh thanks for doing this. I wouldn't have gotten around to it until next week.

File: Assets/MixedRealityToolkit/Utilities/BuildAndDeploy/UwpPlayerBuildTools.cs (hunk 1 above)
Comment (Weasy666) [reply Assets/MixedRealityToolkit/Utilities/BuildAndDeploy/UwpPlayerBuildTools.cs]: IMHO i would also just wrap the method `UpdateAssemblyCSharpProject` in `UwpAppxBuildTools.cs` and the call here in a `#if !UNITY_2019_1_OR_NEWER` preprocessor guard. With that and the preprocessor guard in the layout, the in Untiy 2019 useless code path would just be gone.

Comment (keveleigh) [reply Assets/MixedRealityToolkit/Utilities/BuildAndDeploy/UwpPlayerBuildTools.cs]: Makes sense! I'll update this

**Pull request merged**